payments-core/src/test/java/com/stripe/android/view/CardBrandViewTest.kt

```diff
@@ -60,8 +90,50 @@ internal class CardBrandViewTest {
                 possibleBrands = listOf(CardBrand.CartesBancaires, CardBrand.MasterCard)
             }
 
-            assertThat(cardBrandView.createNetworksParam()?.preferred).isEqualTo(null)
-            assertThat(cardBrandView.merchantPreferredNetworkParams()?.preferred).isEqualTo(null)
+            assertThat(cardBrandView.paymentMethodCreateParamsNetworks()?.preferred).isEqualTo(null)
+            assertThat(cardBrandView.cardParamsNetworks()?.preferred).isEqualTo(null)
+        }
+
+    @Test
+    fun `networkParams shouldn't be present when brand is unknown, cbc ineligible and preferred networks is not set`() =
+        runCardBrandViewTest {
+            cardBrandView.apply {
+                possibleBrands = listOf(CardBrand.CartesBancaires, CardBrand.MasterCard)
+            }
+
+            assertThat(cardBrandView.paymentMethodCreateParamsNetworks()?.preferred).isEqualTo(null)
+            assertThat(cardBrandView.cardParamsNetworks()?.preferred).isEqualTo(null)
+        }
+
+    @Test
+    fun `networkParams should be present when user selects brand and preferred networks is set`() =
+        runCardBrandViewTest {
+            cardBrandView.apply {
+                isCbcEligible = true
+                possibleBrands = listOf(CardBrand.CartesBancaires, CardBrand.MasterCard)
+                handleBrandSelected(CardBrand.MasterCard)
+                merchantPreferredNetworks = listOf(CardBrand.CartesBancaires)
+            }
+
+            assertThat(cardBrandView.paymentMethodCreateParamsNetworks()?.preferred)
+                .isEqualTo(CardBrand.MasterCard.code)
+            assertThat(cardBrandView.cardParamsNetworks()?.preferred)
+                .isEqualTo(CardBrand.MasterCard.code)
+        }
+
+    @Test
+    fun `networkParams should be present when user selects brand, cbc ineligible and preferred networks is set`() =
```

Comment:
Does this represent a real situation? I think a user generally wouldn't be able to select a brand if the merchant is not cbc eligible

Reply:
Yh, it can't happen. Just took it out.